examples/2d/pixel_grid_snap.rs

```diff
@@ -0,0 +1,173 @@
+//! Shows how to create graphics that snap to the pixel grid by rendering to a texture in 2D
+
+use bevy::{
+    prelude::*,
+    render::{
+        camera::RenderTarget,
+        render_resource::{
+            Extent3d, TextureDescriptor, TextureDimension, TextureFormat, TextureUsages,
+        },
+        view::RenderLayers,
+    },
+    sprite::MaterialMesh2dBundle,
+    window::WindowResized,
+};
+
+/// In-game resolution width.
+const RES_WIDTH: u32 = 160;
+
+/// In-game resolution height.
+const RES_HEIGHT: u32 = 90;
+
+/// Default render layers for pixel-perfect rendering.
+/// You can skip adding this component, as this is the default.
+const PIXEL_PERFECT_LAYERS: RenderLayers = RenderLayers::layer(0);
+
+/// Render layers for high-resolution rendering.
+const HIGH_RES_LAYERS: RenderLayers = RenderLayers::layer(1);
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins.set(ImagePlugin::default_nearest()))
+        .insert_resource(Msaa::Off)
+        .add_systems(Startup, (setup_camera, setup_sprite, setup_mesh))
+        .add_systems(Update, (rotate, fit_canvas))
+        .run();
+}
+
+/// Low-resolution texture that contains the pixel-perfect world.
+/// Canvas itself is rendered to the high-resolution world.
+#[derive(Component)]
+struct Canvas;
+
+/// Camera that renders the pixel-perfect world to the [`Canvas`].
+#[derive(Component)]
+struct InGameCamera;
+
+/// Camera that renders the [`Canvas`] (and other graphics on [`HIGH_RES_LAYERS`]) to the screen.
+#[derive(Component)]
+struct OuterCamera;
+
+#[derive(Component)]
+struct Rotate;
+
+fn setup_sprite(mut commands: Commands, asset_server: Res<AssetServer>) {
+    // the sample sprite that will be rendered to the pixel-perfect canvas
+    commands.spawn((
+        SpriteBundle {
+            texture: asset_server.load("pixel/bevy_pixel_dark.png"),
+            transform: Transform::from_xyz(-40., 20., 2.),
+            ..default()
+        },
+        Rotate,
+        PIXEL_PERFECT_LAYERS,
+    ));
+
+    // the sample sprite that will be rendered to the high-res "outer world"
+    commands.spawn((
+        SpriteBundle {
+            texture: asset_server.load("pixel/bevy_pixel_light.png"),
+            transform: Transform::from_xyz(-40., -20., 2.),
+            ..default()
+        },
+        Rotate,
+        HIGH_RES_LAYERS,
+    ));
+}
+
+fn setup_mesh(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+) {
+    commands.spawn((
+        MaterialMesh2dBundle {
```

Comment:
Any particular reason for this to be a mesh2d instead of a sprite?

Reply:
If we're purposely demonstrating that this works with mesh2d, maybe we could pick something other than quad for the shape?

Reply:
I think I wanted to emphasize that any kind of renderable object works, not just a sprite.

Reply:
Makes sense!